src/ReferenceGenerator/Program.cs

```diff
@@ -106,6 +106,13 @@ private static int Main(string[] args)
                 UpdateNuspecFile(nuspecFile, groups, tfms);
                 return 0;
             }
+            catch (UnixNotSupportedException)
```

Comment:
An easy fix for the "access denied" issue on Unix would be to fail fast at the very beginning of Main before trying to do anything.